optimum/intel/neural_compressor/quantization.py

```diff
@@ -535,7 +536,11 @@ def from_pretrained(
             raise ValueError("INC IPEX is currently not supported")
 
         # Load the state dictionary of the model to verify whether the model is quantized or not
-        state_dict = torch.load(state_dict_path)
+        if not cpu_flag: 
+            state_dict = torch.load(state_dict_path)
+        elif cpu_flag:
+            state_dict = torch.load(state_dict_path,map_location='cpu')
+        
```

Comment:
What do you think about always loading the model state dict on the cpu by default ? Would be most likely the most common inference usage + could avoid potential out of memory error (as [in](https://github.com/huggingface/transformers/blob/main/src/transformers/trainer.py#L2062) or [in](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L415) and discussed [in](https://discuss.pytorch.org/t/out-of-memory-error-when-resume-training-even-though-my-gpu-is-empty/30757) and [in](https://discuss.pytorch.org/t/cuda-error-out-of-memory-when-load-models/38011))

Reply:
Yes, that makes sense to me, to load with CPU by default, and to have the option to load with GPU.

Reply:
In most cases, I would say using GPU by default would make more sense, but in the case of Optimum Intel, probably most are using CPU by default at this point.

Reply:
Yes, agreed. By default, it tries to load it on the GPU, which doesn't make sense.

Reply:
Yes always loading and keeping the model on the CPU makes sense then